File: go-main/models/image.go
package models

import "time"

type Image struct {
	Id        int       `json:"id" bson:"_id,omitempty gorm:"primaryKey"`
	URL       string    `json:"url" bson:"url,omitempty"`
	Width     int       `json:"width" bson:"width,omitempty"`
	Height    int       `json:"height" bson:"height,omitempty"`
	Size      int       `json:"size" bson:"size,omitempty"`
	MimeType  string    `json:"mime_type" bson:"mime_type,omitempty"`
	CreateAt  time.Time `json:"created_at" bson:"created_at,omitempty"`
	UpdatedAt time.Time `json:"updated_at" bson:"updated_at,omitempty"`
}

File: admin-web/src/actions/image.ts
import ajax from "./request";

interface Image {
    id: number;
    url: string;
    width: number;
    height: number;
    size: number;
    mime_type: string;
    create_at: string;
    updated_at: string;
}

export function getImageList(): Promise<Image[]> {
    return ajax({
        url: "/image",
        method: "GET",
    }).then((res) => res.data);
}

export function uploadImageToGithubRepo(formData: FormData): Promise<unknown> {
    return ajax({
        url: "/image/publish",
        method: "POST",
        data: formData,
        headers: {
            "Content-Type": "multipart/form-data",
        },
    });
}
